### PR TITLE
New API

### PR DESCRIPTION
**Pull request by rexxars**
This PR introduces a whole new API that I think is much easier to work with and understand.
All the nodes are broken down into small units that are overridable, and I think the terminology is much clearer now.

I'd like more eyes on the API in order to ensure that we get the API right this time and that we can port the same API to the other clients and libraries to provide a uniform way of rendering block content.

From the readme:

## Usage

```js
const React = require('react')
const ReactDOM = require('react-dom')
const BlockContent = require('@sanity/block-content-to-react')
const client = require('@sanity/client')({
  projectId: '<your project id>',
  dataset: '<some dataset>',
  useCdn: true
})

const serializers = {
  types: {
    code: props => (
      <pre data-language={props.node.language}>
        <code>{props.node.code}</code>
      </pre>
    )
  }
}

client.fetch('*[_type == "article"][0]').then(article => {
  ReactDOM.render(
    <BlockContent blocks={article.body} serializers={serializers} />,
    document.getElementById('root')
  )
})
```

## PropTypes

- `className` - When more than one block is given, a container node has to be created. Passing a `className` will pass it on to the container. Note that if only a single block is given as input, the container node will be skipped.
- `serializers` - Specifies the React components to use for rendering content. Merged with default serializers.
- `serializers.types` - Serializers for block types, see example above
- `serializers.marks` - Serializers for marks - data that annotates a text child of a block. See example usage below.
- `serializers.list` - React component to use when rendering a list node
- `serializers.listItem` - React component to use when rendering a list item node
- `imageOptions` - When encountering image blocks, this defines which query parameters to apply in order to control size/crop mode etc.

In addition, in order to render images without materializing the asset documents, you should also specify:

- `projectId` - The ID of your Sanity project
- `dataset` - Name of the Sanity dataset containing the document that is being rendered

### Rendering custom marks

```js
const input = [{
  _type: 'block',
  children: [{
    _key: 'a1ph4',
    _type: 'span',
    marks: ['s0m3k3y'],
    text: 'Sanity'
  }],
  markDefs: [{
    _key: 's0m3k3y',
    _type: 'highlight',
    color: '#E4FC5B'
  }]
}]

const highlight = props => {
  return (
    <span style={{backgroundColor: props.mark.color}}>
      {props.children}
    </span>
  )
}

<BlockContent
  blocks={input}
  serializers={{marks: {highlight}}}
/>
```

### Specifying image options

```js
<BlockContent
  blocks={input}
  imageOptions={{w: 320, h: 240, fit: 'max'}}
  projectId="myprojectid"
  dataset="mydataset"
/>
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+language: node_js
```
Comment (bjoerge) [inline .travis.yml]: Do you know if its possible to make Travis CI run on individual packages in our monorepo?

Comment (bjoerge) [reply .travis.yml]: (Sorry for the irrelevant comment, I thought this package was actually _in_ our monorepo. Still wondering if you know its possible though 😇)

Comment (rexxars) [reply .travis.yml]: Sort of. I was thinking we might bootstrap and then specify explicitly which packages to run tests on as `cd packages/@sanity/<some-pkg> && npm test`. If you catch my drift.

**Comment by thomax** [comment]
This is a huge improvement, looking very much forward to use this!

If I have one gripe, it's having to include `projectId` and `dataset` as props for `<BlockContent../>`. I see why they're needed, but I'm wondering if there is another way to access them, instead of pumping them in ever time BlockContent should render an image?

**Comment by rexxars** [comment]
@thomax I agree, but I couldn't come up with a better design. I guess you could pass an instance of an instantiated sanity client, but not sure if that feels better or worse?

Alternatively, you can provide your own serializer for image, or materialize images ahead of time which will let you handle this without specifying them.

Any suggestions on how we could handle this more smoothly is very welcome.

**Comment by thomax** [comment]
@rexxars Yup, I see the problem. I guess something like https://www.npmjs.com/package/react-global-configuration could be done, and then have BlockContent load projectId and dataset from there. But if you have multiple datasets from which to grab images you have to name them somehow, and then we're full circle. Sooo, even though most use cases will be single dataset, it might not be worth contaminating the BlockContent component with the additional complexity of yucky global config.

**Comment by rexxars** [comment]
Personally I would just create a reusable `MyBlockContent`-component with the project config, then pass on the props:

MyBlockContent.js:
```js
const BlockContent = require('block-content-to-react')
const config = require('../my-thing/config')

const MyBlockContent = props =>
  <BlockContent
    projectId={config.projectId}
    dataset={config.dataset}
    {...props}
  />

module.exports = MyBlockContent
```

**Pull request merged**